### PR TITLE
Add support for Lehe and LehePML solvers in 2D

### PR DESCRIPTION
**Pull request by sbastrakov**
This combination was already mostly enabled by previous PRs. Add special handling of averaging along z. Remove the guards that conditionally enable Lehe for 3D only. Cherenkov-free direction Z does not make sense in 2D, this is checked at compile time.

### CODE REVIEW

File: include/picongpu/fields/MaxwellSolver/Lehe/Derivative.hpp [hunk 1]
```diff
@@ -120,13 +120,16 @@ namespace lehe
             float_64 const coeff = stepRatio * math::sin(
                 pmacc::math::Pi< float_64 >::halfValue *
                 float_64( SPEED_OF_LIGHT ) * float_64( DELTA_T ) /
-                float_64( cellSize[ T_direction ] )
+                float_64( cellSize[ dir0 ] )
```
Comment (sbastrakov) [inline include/picongpu/fields/MaxwellSolver/Lehe/Derivative.hpp:123]: Note: this is unrelated, `dir0` is same as `T_direction`, just for more uniformity inside this function.

**Comment by sbastrakov** [comment]
@psychocoderHPC the tests passed, could you review?

**Comment by psychocoderHPC** [comment]
I removed the lable `user input` because this change is not effecting the input files directly. I would keep the flag to point to changes where param files are changed.
